# <p style="text-align: right; direction: rtl; float: right;">תרגיל בית להגשה - סמסטר ב' תשפ"ב</p>

### <p style="text-align: right; direction: rtl; float: right;">מגיש:</p>

# Shiraz Hadad, ID 207256835

<p style="text-align: right; direction: rtl; float: right; clear: both;">
בשעה טובה הגיע זמנו של תרגיל הבית.<br>
את רוב השאלות אתם כבר יודעים לפתור. מה שלא - חלק גדול נלמד בתרגולים הקרובים והשאר נמצא בקבצים שהעלנו המתרגלים למודל.<br><br>
מטרת תרגיל הבית הזה הוא לתת לכם להתנסות בפונקציות סטטסיטיות בפייתון, ושבעזרת ה' בכל פעם שתצטרכו לחשב משהו בסטטיסטיקה, בין אם בקורסים בהמשך התואר ובין אם בשוק העבודה - <mark>תרגישו בנוח לפתוח קובץ פייתון, ולקבל את התוצאות בקלות וביעילות.</mark><br>
</p>
<p style="text-align: right; direction: rtl; float: right; clear: both;">
את התרגיל תעשו ביחידים, ויש לפתור אותו בעצמכם.<br>
התרגיל באמת פשוט, השתדלתי מאוד לתת כמה שיותר ערך סטטיסטי וכמה שפחות להתמקד בצד הפייתוני של העבודה.<br>
שימו לב שאחרי התשובות, בעיקר בחלקים של מבחני ההשערות - <strong>עליכם להסביר במילים</strong> את תוצאת המבחן שביצעתם ומה מסקנתכם!! <mark>ההסבר הוא החלק החשוב ביותר.</mark><br><br>
לאורך הקובץ אתם יכולים להוסיף תאים כראות עיניכם.<br><br>
אין צורך לכתוב פונקציות משלכם, הספריות הסטטיסטיות שלמדנו צריכות להספיק בהחלט. הרעיון הוא שתדעו להשתמש בפונקציות, ולא שתדעו לתכנת.<br>
מי שרוצה לבצע חישוב בצורה "ידנית" בפייתון כדי לוודא שהשתמש בנוסחה המתאימה לחומר שלמדנו - אין בעיה. אבל אין חובה כזו.<br>
</p>

### <p style="text-align: right; direction: rtl; float: right;">לפני הכל - לא לשכוח לייבוא את הספריות:</p>

In [2]:
import numpy as np
import pandas as pd
import scipy.stats as st 
import matplotlib.pyplot as plt
import statsmodels.api as sm
import statsmodels.stats.weightstats as sws
from statsmodels.stats.proportion import proportion_confint

<br><br><br>
## <p style="text-align: right; direction: rtl; float: right;">שאלה א.</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>א 1 .</strong><br>
צרו וקטור בשם  x_unif עם 13 ערכים מהתפלגות אחידה בדידה בין 1 ל-6, והציגו אותו.<br>

In [3]:
min = 1
max = 6
SampleSize = 13
x_unif = np.random.uniform(min, max,SampleSize)
x_unif

array([4.94187371, 5.98680366, 3.0237037 , 5.91103194, 5.05032574,
       3.4784764 , 1.65508649, 5.64719449, 5.96453105, 1.30934157,
       2.32660232, 2.37751902, 3.01630618])

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>א 2 .</strong><br>
צרו וקטור בשם x_stand עם 8 ערכים מהתפלגות נורמלית סטנדרטית, והציגו אותו.<br>

In [4]:
mu = 0
sigma = 1 
SampleSize = 8 
x_stand = np.random.normal(mu,sigma,SampleSize)
x_stand

array([ 1.38288222,  1.10298777, -0.75152966,  0.39030999,  0.10586335,
        0.65006518,  1.41384981,  0.51264186])

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>א 3 .</strong><br>
צרו וקטור  עם 100 ערכים מהתפלגות נורמלית עם תוחלת 100 וסטיית תקן 7, והציגו אותו.<br>

In [5]:
mu = 100
sigma = 7 
SampleSize = 100 
y_stand = np.random.normal(mu,sigma,SampleSize)
y_stand

array([ 99.60589979, 107.18671446,  87.53668867,  93.0125512 ,
        95.00839241,  98.15810577, 108.73929814,  97.79124093,
        97.06863493,  88.41260151,  92.35857554,  94.1870386 ,
        95.48504229, 102.78104389, 103.96573691,  96.98376148,
        93.77557813,  99.03904309,  95.25790368,  99.11562436,
       106.39385621, 100.03036838,  97.21657767, 112.81420037,
        98.52501668,  95.67462937,  98.8023945 , 101.67866556,
       106.74423395,  92.47917646, 100.47277644,  96.48752466,
        93.20491554,  88.24632085, 107.60980129, 100.78862604,
       101.11944806, 104.84204545,  90.91467569, 103.0682873 ,
       103.84914979,  95.52832723, 100.05379754,  89.33132127,
        97.27896806, 105.33279401, 108.27024087, 106.97665165,
       103.32929129,  98.07597965,  89.28560723,  99.22661213,
        90.59258902, 103.02124145,  99.09887066, 105.36322244,
       111.73202144,  97.89325622, 109.81452699,  96.96474505,
       104.89497679,  89.89167576, 109.61783018, 110.10

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>א 4 .</strong><br>
צרו וקטור בשם x_pois עם 12 ערכים מהתפלגות פואסון עם למדא 7.3, והציגו אותו.<br>

In [6]:
lamda = 7.3
x_pois = np.random.poisson(lamda,size = 12)
x_pois

array([ 7,  9,  7, 13,  5,  5,  7,  8,  8, 12,  9,  9])

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>א 5 .</strong><br>
צרו וקטור בשם x_expon עם 3,000 ערכים מהתפלגות מעריכית עם למדא 7.3, ואל תציגו אותו, רק חשבו את הממוצע שלו ואת סטיית התקן המתוקנת. (שימו לב שתוחלת התפלגות מעריכית היא 1 חלקי למדא, ולא למדא).<br>

In [7]:
mu = 7.3
SampleSize = 3000
x_expon = np.random.exponential(scale = mu, size = 3000)

In [8]:
np.mean(x_expon)

7.3744210724933845

In [9]:
st.sem(x_expon,axis = 0,ddof=0)

0.1305876675786181

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>א 6 .</strong><br>
צרו וקטור בשם x_binom עם 400 ערכים מהתפלגות בינומית, עם 7 ניסיונות וסיכוי להצלחה 0.15, והציגו אותו.<br>

In [10]:
k = 7 
p = 0.15 
n = 400 
x_binom = np.random.binomial(k, p, n)
x_binom

array([3, 1, 3, 2, 2, 1, 1, 1, 0, 0, 3, 3, 1, 0, 2, 0, 1, 4, 3, 0, 2, 2,
       0, 0, 0, 2, 0, 1, 1, 0, 1, 1, 1, 1, 2, 3, 0, 2, 2, 2, 0, 1, 0, 1,
       0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 2, 0, 0, 2, 0, 0, 0, 1, 0, 0, 0,
       1, 2, 2, 0, 1, 1, 1, 2, 1, 1, 2, 2, 2, 0, 1, 3, 2, 2, 3, 0, 0, 0,
       2, 1, 2, 1, 1, 0, 1, 2, 2, 2, 1, 1, 2, 3, 2, 0, 0, 1, 0, 2, 0, 3,
       2, 1, 0, 1, 1, 1, 2, 2, 2, 0, 1, 1, 1, 2, 1, 0, 0, 0, 1, 1, 0, 0,
       2, 1, 1, 1, 3, 1, 0, 2, 0, 2, 1, 1, 0, 1, 2, 1, 0, 4, 0, 0, 1, 2,
       4, 2, 0, 0, 1, 1, 1, 0, 0, 2, 2, 2, 1, 1, 0, 1, 0, 1, 2, 2, 1, 0,
       0, 0, 1, 0, 1, 0, 0, 2, 3, 2, 2, 1, 1, 2, 1, 0, 3, 1, 2, 2, 0, 1,
       2, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 0, 0, 0, 2, 0, 0, 1, 0, 0, 2, 1,
       1, 1, 3, 2, 0, 1, 2, 1, 1, 1, 0, 1, 1, 1, 0, 0, 4, 1, 1, 1, 2, 2,
       3, 1, 3, 1, 1, 0, 1, 2, 3, 1, 1, 1, 0, 1, 1, 2, 2, 0, 0, 0, 1, 1,
       0, 0, 1, 0, 2, 3, 0, 1, 2, 2, 0, 1, 1, 1, 1, 2, 1, 2, 1, 0, 0, 0,
       1, 1, 2, 0, 1, 1, 3, 0, 1, 1, 0, 1, 0, 0, 1,

<br><br><br>
## <p style="text-align: right; direction: rtl; float: right;">שאלה ב.</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>את הסעיפים הבאים יש לחשב באמצעות פונקציות בפייתון.</strong><br>
אם צריך ניתן לחבר בין פונקציות או לכתוב "אחד פחות" וכדו'<br>


<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 1 .</strong><br>
מהו האחוזון ה-95 בהתפלגות נורמלית סטנדרטית?<br>

In [11]:
st.norm.ppf(0.95)

1.6448536269514722

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 2 .</strong><br>
מהו האחוזון ה-50 בהתפלגות נורמלית עם תוחלת 70, וסטיית תקן 10?<br>

In [12]:
st.norm.ppf(0.5, 70,10)

70.0

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 3 .</strong><br>
מה ערך ההתפלגות המצטברת עד 1- בהתפלגות נורמלית סטנדרטית?<br>

In [13]:
st.norm.cdf(-1)

0.15865525393145707

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 4 .</strong><br>
כמה אחוז מהאוכלוסיה נמצאים בהתפלגות נורמלית בטווח שבין סטיית תקן אחת מתחת לתוחלת, לבין סטיית תקן אחת מעל התוחלת?<br>

In [14]:
x = st.norm.cdf(11,8,1)
y = st.norm.cdf(9,8,1)

In [15]:
population_percentage = (x - y) * 100
population_percentage

15.730535589982697

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 5 .</strong><br>
מהו ערך פונקציית הצפיפות של התפלגות נורמלית סטנדרטית בנקודה x=0?<br>

In [16]:
st.norm.pdf(0)

0.3989422804014327

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 6 .</strong><br>
מהו ערך פונקציית הצפיפות של התפלגות מעריכית עם למדא=1, בנקודה x=1?<br>

In [17]:
st.expon.pdf(1,0,1)

0.36787944117144233

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 7 .</strong><br>
מהי ההסתברות המצטברת בהתפלגות מעריכית (למדא=1) בנקודה x=1?<br>

In [18]:
st.expon.cdf(1,0,1)


0.6321205588285577

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 8 .</strong><br>
מה ההסתברות לקבל 3 הצלחות בהתפלגות בינומית עם 7 ניסיונות וסיכוי להצלחה 0.15?<br>

In [19]:
st.binom.pmf(3, 7, 0.15,loc = 0)

0.061661988281249934

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 9 .</strong><br>
מה ההסתברות לקבל לפחות 3 הצלחות בהתפלגות בינומית עם 7 ניסיונות וסיכוי להצלחה 0.15?<br>

In [20]:
st.binom.sf(2,7,0.15)

0.07376516015625

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 10 .</strong><br>
מה ההסתברות שיגיעו בדיוק 10 מכוניות לתחנת דלק ביחידת זמן, אם קצב ההגעה מתפלג פואסונית עם למדא=10?<br>

In [21]:
st.poisson.pmf(10,10)

0.12511003572113372

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 11 .</strong><br>
מה ההסתברות שיגיעו לכל היותר 10 מכוניות לתחנת דלק ביחידת זמן, אם קצב ההגעה מתפלג פואסונית עם למדא=10?<br>

In [22]:
st.poisson.cdf(10,10)

0.5830397501929852

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 12 .</strong><br>
מהו האחוזון ה-92 בהתפלגות t עם 7 דרגות חופש?<br>

In [23]:
st.t.ppf(q=0.92,df=7)

1.5717661843733877

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 13 .</strong><br>
בהתפלגות t, כמה שטח מכוסה בטווח שבין 1- ל1? (תבחרו כמה דרגות חופש שבא לכם)<br>

In [24]:
x= st.t.cdf(x = 1,df = 10, loc = 0, scale = 1)
y= st.t.cdf(x = -1,df = 10, loc = 0, scale = 1)
x-y

0.6591068676979399

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 14 .</strong><br>
מה ההסתברות לקבל פחות מ-20 בדגימה אקראית מהתפלגות חי בריבוע עם 20 דרגות חופש?<br>

In [25]:
st.chi2.cdf(20,20)

0.5420702855281478

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 15 .</strong><br>
לקחנו 2 משתנים שמתפלגים חי-בריבוע וחילקנו אותם זה בזה. במונה היה משתנה עם 71 ד"ח, ובמכנה היה משתנה עם 26 דרגות חופש. מה ההסתברות לקבל ערך גדול מ-1.5? (שימו לב איזה התפלגות זו?)<br>

In [26]:
1-st.f.cdf(1.5, 71, 26)

0.12453707113979473

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ב 16 .</strong><br>
בהתפלגות F, עם 12 ד"ח במונה ו-20 ד"ח במכנה. מה ההסתברות לקבל ערך קטן מ-0.3937?<br>

In [27]:
st.f.cdf(0.3937,12,20)

0.050246693977039694

<br><br><br>
## <p style="text-align: right; direction: rtl; float: right;">שאלה ג.</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
קיראו את הקובץ nba_salaries.<br>
ניתן להעזר בקטע קוד הבא, הכניסו את הנתיב לתיקיה שבה שמרתם את הקובץ. שימו לב לכיוון הסלשים.<br><br>      
</p>
<p style="text-align: right; direction: rtl; float: right; clear: both;">
בקובץ מפורטים: שמות השחקנים, הקבוצה אליה הם משתייכים, מספר הגופיה שלהם, עמדה על המגרש, גיל, גובה, משקל, הקולג' בו למדו ושיחקו, והמשכורת של השחקן.<br>

In [28]:
PATH = "file:///C:/Users/שירז/סטטיסטיקה/מטלה%20פייתון/nba_salaries.csv"
nba_salaries = pd.read_csv(PATH)
nba_salaries.head()

URLError: <urlopen error [WinError 3] The system cannot find the path specified: 'C:\\Users\\שירז\\סטטיסטיקה\\מטלה פייתון\\nba_salaries.csv'>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
כדי לא להסתבך, נזרוק את כל השורות שיש בהן NaN.<br>
בעתיד אתם תשתמשו בחומר של סטטיסטיקה כדי לנהל בצורה יותר חכמה ערכים חסרים, אבל זה מחוץ לסקופ של הקורס שלנו.<br>

In [ ]:
nba_salaries = nba_salaries.dropna()
nba_salaries.head()

<p style="text-align: right; direction: rtl; float: right; clear: both;">
כדי לבחור למשל את העמודה של המשכורות אנחנו יכולים להשתמש בפקודה הבאה:<br>

In [31]:
nba_salaries['Salary']

0      7730337.0
1      6796117.0
3      1148640.0
6      1170960.0
7      2165160.0
         ...    
449    1348440.0
451     981348.0
452    2239800.0
453    2433333.0
456     947276.0
Name: Salary, Length: 364, dtype: float64

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ג 1 .</strong><br>
צרו רווח סמך ברמת ביטחון 95% לתוחלת השכר של שחקן NBA.<br>

#### לא נתונה לנו השונות וגודל המדגם גדול מ-30, לכן נשתמש במבחן t

In [32]:
st.t.interval(alpha=0.95, df=nba_salaries['Salary'].count()-1, loc=np.mean(nba_salaries['Salary']), scale=st.sem(nba_salaries['Salary']))

(4092603.1028615795, 5148019.039995564)

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ג 2 .</strong><br>
נניח שיצא שהרווח הוא: [5,000,000 , 4,000,000]. האם ניתן לומר כי 95% מהשחקנים ב-NBA מרוויחים בין 4 ל-5 מיליון דולר? הסבירו.<br>

### כאשר אנו עושים רווח סמך אנחנו מקבלים את הטווח ברמת ביטחון מסויימת (לדוגמה 95%) שהמשמעות הדבר זה שאנחנו נפלול בטווח שקיבלנו ברמת ביטחון 95% , אבל על מנת לקבל את אחוז השחקנים שיפלו בתוך הטווח עלינו לעשות רווח סמך לטולרנס ואז אם קיבלנו 95% אז ניתן להגיד ש 95% מהשחקנים יכולים ליפול בטווח הזה

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ג 3 .</strong><br>
מהו האומדן הנקודתי לפרופורציית מספר הסנטרים בליגה?<br>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
נוכל להשתמש בפקודה הזו כדי לספור כמה שחקנים יש בדאטה-בייס שלנו בעמדת הסנטר:<br>

In [33]:
num_of_centers = nba_salaries['Position'][nba_salaries['Position']=='C'].count()
num_of_centers

49

<p style="text-align: right; direction: rtl; float: right; clear: both;">
ובפקודה הזו כדי לספור כמה שחקנים יש סך הכל בדאטה-בייס שלנו:<br>

In [39]:
players = len(nba_salaries)
players

364

In [40]:
x = num_of_centers / players
x

0.1346153846153846

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ג 4 .</strong><br>
צרו רווח סמך לפרופורציית מספר הסנטרים בליגה (95%). <br>

In [41]:
proportion_confint(count=49,    # Number of centers
                   nobs=364,    # Number of players
                   alpha=0.05)

(0.09955236356343752, 0.1696784056673317)

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ג 5 .</strong><br>
מהי המשכורת הממוצעת לשחקן NBA?<br>

In [42]:
np.mean(nba_salaries['Salary'])

4620311.071428572

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ג 6 .</strong><br>
בידקו את ההשערה כי תוחלת השכר של שחקנים בעמדת הסנטר גבוהה יותר מתוחלת השכר הכללית. מה מסקנתך ברמת מובהקות של 5%? של 10%?<br>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
בצורה הבאה נוכל לקחת וקטור נתונים של כל המשכורות של השחקנים בעמדת הסנטר:<br>

In [32]:
salary_of_centers = nba_salaries['Salary'][nba_salaries['Position']=='C']
salary_of_centers.head()

7      2165160.0
10     2569260.0
14     2616975.0
23    19689000.0
27      947276.0
Name: Salary, dtype: float64

In [44]:
p = nba_salaries['Salary']
l = nba_salaries['Salary'][nba_salaries['Position']=='C']
st.ttest_ind(l,p, alternative='greater')

Ttest_indResult(statistic=1.4446580964036155, pvalue=0.07465793588804467)

### מכיוון שקיבלנו שפיבאליו הינו 0.07, ברמת ביטחון של 95% אנו נקבל את השערת האפס, אך ברמת ביטחון של 90% אנו נדחה אותה

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ג 7 .</strong><br>
מהו המשקל הממוצע של שחקן NBA?<br>

In [47]:
np.mean(nba_salaries['Weight'])

219.78571428571428

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ג 8 .</strong><br>
עלתה טענה כי תוחלת משקלם של שחקנים בעמדת הפוינט-גארד (PG) שונה מתוחלת המשקל של כלל השחקנים בליגה. בידקו טענה זו. מהי מובהקות התוצאה? האם ניתן לומר האם משקלם גבוה או נמוך יותר מהמשקל של כלל האוכלוסיה?<br>

In [48]:
all_players_weight = nba_salaries['Weight']
PG_players_weight = nba_salaries['Weight'][nba_salaries['Position']=='PG']

st.ttest_ind(PG_players_weight,all_players_weight,alternative='two-sided')

Ttest_indResult(statistic=-10.146131415241898, pvalue=7.237781882699931e-22)

### לאור זאת שקיבלנו שערך הפיבאליו שואף לאפס נוכל להגיד שיש שוני בין המדגם לאוכלוסיה 

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ג 9 .</strong><br>
בידקו את הטענה כי יש שוני בין תוחלת משכורת שחקני לוס אנג'לס לייקרס לתוחלת משכורת שחקני הבוסטון סלטיקס. מהי מסקנתך?<br>

In [49]:
f = nba_salaries['Salary'][nba_salaries['Team']=='Boston Celtics']
v = nba_salaries['Salary'][nba_salaries['Team']=='Los Angeles Lakers']
st.ttest_ind(f,v,alternative='two-sided')

Ttest_indResult(statistic=0.1374821899368491, pvalue=0.8919001800358959)

### לאור זאת שקיבלנו ערך גדול מאוד בפיבאליו שכמעט בכל רמת ביטחון נקבל את השערת האפס

<br><br><br>
## <p style="text-align: right; direction: rtl; float: right;">שאלה ד.</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ד 1 .</strong><br>
חברת כימיקלים הציגה דשן מיוחד שלטענתם יכול להגדיל משמעותית את היבול. להלן נתונים שהציגה החברה:<br>
בדוק האם טענת החברה נכונה. מהי מסקנתך?<br>
<br>

In [50]:
fertilizered = pd.DataFrame({"num_of_field": [1, 2, 3, 4, 5, 6, 7],
                           "before": [198, 236, 362, 497, 203, 530, 429],
                          "after": [271, 239, 412, 592, 310, 488, 512]})
fertilizered

num_of_field  before  after
0             1     198    271
1             2     236    239
2             3     362    412
3             4     497    592
4             5     203    310
5             6     530    488
6             7     429    512

<p style="text-align: right; direction: rtl; float: right; clear: both;">
בידקו האם טענת החברה נכונה. מהי מסקנתך?<br>
<br>

In [57]:
before = fertilizered.iloc[:,1]
after = fertilizered.iloc[:,2]
st.ttest_rel(a = after, b = before,alternative='greater')

Ttest_relResult(statistic=2.581471790797932, pvalue=0.02084344831321906)

## ברמת ביטחון של 99% ניתן להגיד שהטענה אינה נכונה (נקבל את השערת האפס) אך ברמת ביטחון של 95% ניתן להגיד שהטענה נכונה (נדחה את השערת האפס)

<br><br><br>
## <p style="text-align: right; direction: rtl; float: right;">שאלה ה.</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ה 1 .</strong><br>
במדגם של 194 סטודנטים אנו רוצים לבדוק האם ההתפלגות של המגזרים שלהם שווה להתפלגות באוכלוסיה כולה.<br>
מצאנו בויקיפדיה עולה כי בישראל יש 73.9% יהודים, 21.1% ערבים ו-5% תחת הקטגוריה 'אחר'.<br>
במדגם האקראי שלקחנו היו 132 יהודים, 42 ערבים ו-20 המוגדרים 'אחר'.<br>

In [59]:
students_sample = [132, 42, 20]
expeced = [0.739*194, 0.211*194, 0.05*194]

<p style="text-align: right; direction: rtl; float: right; clear: both;">
בצעו מבחן חי-בריבוע לטיב התאמה ובדקו האם אכן ייתכן שהתפלגות אוכלוסיית הסטודנטים זהה להתפלגות האוכלוסיה. מהי מסקנתכם?<br>
<br>

In [60]:
st.chisquare(f_obs=students_sample,f_exp=expeced)

Power_divergenceResult(statistic=11.86596609228638, pvalue=0.0026505634309167323)

## לאחר שקיבלנו ערך סטטיסטי 11.8 ופיואליו קטן מ-0.01 ניתן לדחות את השערת האפס שאומרת שההתפלגות אינה זהה

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ה 2 .</strong><br>
באיצטדיון מרוצי סוסים ישנם 8 מסלולים. במשך יומיים של התחרות נערכו 240 מרוצים כאשר כל סוס רץ בכל מקצה במסלול אחר באופן רנדומלי לחלוטין. <br>אלברט ספר כמה זכיות היו בכל מסלול וקיבל את הנתונים הבאים:<br>
[22, 46, 28, 40, 30, 30, 24, 20]<br>
האם אכן המסלולים השונים אינם משפיעים על סיכויי הזכיה?<br>
<br>

In [62]:
observed = [22, 46, 28, 40, 30, 30, 24, 20]
expected = [30, 30, 30, 30, 30, 30, 30, 30]
st.chisquare(observed,expected)

Power_divergenceResult(statistic=18.666666666666664, pvalue=0.009298218378412404)

## לאחר שקיבלנו ערך סטטיסטי 18.6 ופיואליו קטן מ 0.01 ניתן לדחות את השערת האפס שאומרת שהם אינם משפיעים, ולכן הם כן משפיעים

<p style="text-align: right; direction: rtl; float: right; clear: both;">
<strong>ה 3 .</strong><br>
במפעל רשמו את מספר הפגומים והתקינים בכל משמרת (בוקר, אחר-צהריים ולילה), התיעוד נמצא בקובץ בשם defects.<br>
קראו את הנתונים ובידקו האם יש קשר בין סוג המשמרת לבין מספר הפגומים והתקינים. מהי מסקנתכם?
<br>

In [68]:
PATH = "file:///C:/Users/שירז/סטטיסטיקה/מטלה%20פייתון/Defects.csv"
defects = pd.read_csv(PATH, header=0, index_col=0)
defects

morning  afternoon  night
defective       30         55     75
proper          70         35     35

In [69]:
st.chi2_contingency(defects)

(33.806818181818194,
 4.5597675805207196e-08,
 2,
 array([[53.33333333, 48.        , 58.66666667],
        [46.66666667, 42.        , 51.33333333]]))

## לאחר שקיבלנו ערך סטטיסטי 33.8 ופיואליו שואף לאפס ניתן לדחות את השערת האפס שאומרת שאין קשר בין הנתונים

<br><br><br>
## <p style="text-align: right; direction: rtl; float: right;">כל הכבוד!</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
אם הגעת עד לכאן, אנחנו בטוחים שיש לך רצון עז להצליח!!<br>
אנחנו מקווים שאימצתם את פייתון ככלי נוח ויעיל לביצוע חישובים סטטיסטיים.
<br>
    
<br><br><br>
## <p style="text-align: right; direction: rtl; float: right;">בהצלחה!!!</p>